### PR TITLE
Remove tests for, and test the removal of, context menus

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/2742.

Bugs to file:

- Firefox to remove everything
- Edge to remove menu.type
- Chrome to remove onshow

@sideshowbarker, there are some conformance checker tests for menuitem, but I think they are testing invalid cases which stay invalid; would appreciate your help checking those.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/240101257)

# Firefox (nightly) #
Testing web-platform-tests at revision f2123bb9dc7fe4d22aae271ae1fd6b21ff8f337f
Using browser at version BuildID 20170606100219; SourceStamp 58ce95bc58ce4ba200413c8bed87786dccf3d105
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/240101258)

# Sauce (safari) #
Testing web-platform-tests at revision f2123bb9dc7fe4d22aae271ae1fd6b21ff8f337f
Using browser at version 10.0
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/240101266)

# Chrome (unstable) #
Testing web-platform-tests at revision f2123bb9dc7fe4d22aae271ae1fd6b21ff8f337f
Using browser at version 60.0.3112.20 dev
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/240101267)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision f2123bb9dc7fe4d22aae271ae1fd6b21ff8f337f
Using browser at version 14.14393
Starting 10 test iterations

File: html/semantics/interactive-elements/contextmenu-historical.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    assert_false(prop in menuitem, `menuitem.${prop} must not be present`);
+  }
+}, "HTMLMenuItemElement must not be not present");
+
+test(() => {
+  for (const location of [window, HTMLElement.prototype, SVGElement.prototype, Document.prototype]) {
+    assert_false("onshow" in location,
+      `${location.constructor.name} must not have a property "onshow"`);
+  }
+  assert_false("onshow" in Element.prototype, `Element.prototype must not contain an "onshow" property`);
```
Comment (zcorpan) [inline html/semantics/interactive-elements/contextmenu-historical.html]: Why is this not in the array instead?

Maybe also test HTMLDocument.prototype.

File: html/semantics/interactive-elements/contextmenu-historical.html [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+test(() => {
+  assert_false("RelatedEvent" in window);
+}, "RelatedEvent must not be present");
+
+test(() => {
+  assert_false("contextMenu" in HTMLElement.prototype,
+    "HTMLElement's prototype must not have a property \"contextmenu\"");
+  assert_false("contextMenu" in document.createElement("div"),
+    "A div must not have a property \"contextmenu\"");
```
Comment (zcorpan) [inline html/semantics/interactive-elements/contextmenu-historical.html]: Uppercase M in contextmenu

File: html/semantics/interactive-elements/contextmenu-historical.html [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+test(() => {
+  assert_false("type" in menu);
+}, "menu.type must not be present");
+
+test(() => {
+  assert_false("label" in menu);
+}, "menu.label must not be present");
+
+test(() => {
+  console.log(document.querySelectorAll("menuitem:enabled"));
```
Comment (zcorpan) [inline html/semantics/interactive-elements/contextmenu-historical.html]: remove

File: html/semantics/interactive-elements/contextmenu-historical.html [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    // Not everyone has implemented :default as of the time of this writing.
+    if (e.name !== "SyntaxError") {
+      throw e;
+    }
+  }
+}, ":default must not match menuitems");
+
+test(() => {
+  assert_equals(getComputedStyle(menu).display, "block");
+}, "The user-agent stylesheet must type=\"context\" menus as block display");
```
Comment (zcorpan) [inline html/semantics/interactive-elements/contextmenu-historical.html]: This sentence is missing some word

File: html/semantics/interactive-elements/contextmenu-historical.html [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+test(() => {
+  assert_false("contextMenu" in HTMLElement.prototype,
+    "HTMLElement's prototype must not have a property \"contextmenu\"");
+  assert_false("contextMenu" in document.createElement("div"),
+    "A div must not have a property \"contextmenu\"");
+}, "el.contextMenu must not be present");
+
+test(() => {
+  assert_false("type" in menu);
```
Comment (zcorpan) [inline html/semantics/interactive-elements/contextmenu-historical.html]: Also test that setting type does not set a content attribute (and similarly for the others).

**Comment by zcorpan** [comment]
I fixed the conformance checker tests and filed https://github.com/validator/validator/issues/526

**Review by zcorpan — APPROVED** [review]

**Comment by sideshowbarker** [comment]
> I fixed the conformance checker tests and filed validator/validator#526

thanks

**Review by sideshowbarker — APPROVED** [review]

**Comment by jgraham** [comment]
There doesn't seem to be any documentation about why this was merged despite failing the test jobs on all browsers?

**Pull request merged**